src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -442,11 +486,23 @@ ss::future<subject_schema> sharded_store::get_subject_schema(
           return s.get_schema_definition(id).value();
       });
 
-    co_return subject_schema{
-      .schema = {sub, std::move(def)},
-      .version = v_id.version,
-      .id = v_id.id,
-      .deleted = v_id.deleted};
+    try {
+        auto canonical = co_await make_canonical_schema({sub, std::move(def)});
```

Comment:
It would be nice to accept a `nomalize` param into this function (`schema_getter::get_subject_schema` can just wrap this and pass `no`), and forward that param here.

Then `has_schema` can be simplified and the work reduced.

See acdaa45d80ecb2f35697baa92f80e506ff82390e